rmw_connext_shared_cpp/src/event.cpp

```diff
@@ -0,0 +1,61 @@
+// Copyright 2015-2017 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include "rmw/error_handling.h"
+#include "rmw/impl/cpp/macros.hpp"
+
+#include "rmw_connext_shared_cpp/types.hpp"
+#include "rmw_connext_shared_cpp/event.hpp"
+#include "rmw_connext_shared_cpp/event_converter.hpp"
+#include "rmw_connext_shared_cpp/connext_static_event_info.hpp"
+
+
+rmw_ret_t
+__rmw_take_event(
+  const char * implementation_identifier,
+  const rmw_event_t * event_handle,
+  void * event_info,
+  bool * taken)
+{
+  // pointer error checking here
+  RMW_CHECK_ARGUMENT_FOR_NULL(event_handle, RMW_RET_INVALID_ARGUMENT);
+  RMW_CHECK_TYPE_IDENTIFIERS_MATCH(
+    event handle,
+    event_handle->implementation_identifier,
+    implementation_identifier,
+    return RMW_RET_ERROR);
+
+  rmw_ret_t ret_code = RMW_RET_UNSUPPORTED;
+
+  // check if we support the input event type
+  if (is_event_supported(event_handle->event_type)) {
```

Comment:
If not, should you not set an appropriate error message?

Reply:
Will do.